src/main/java/org/jspecify/meta/Implies.java

```diff
@@ -24,19 +24,82 @@
+ *
+ * <p>We will refer to this example throughout this documentation:
+ *
+ * <pre>{@code
+ * public @interface One {}
+ *
+ * // Additional meaning of `@Two`, if any, specified here.
+ * @Implies(One.class)
+ * public @interface Two {}
+ *
+ * // These might be classes, fields, methods, etc.
+ * @One declaration_a
+ * @Two declaration_b
+ * @One @Two declaration_c
+ * }</pre>
+ *
+ * <p>In this example, the meaning conveyed by the annotations is exactly the same for {@code
+ * declaration_b} and {@code declaration_c}; there, {@code @One} is considered redundant (but
+ * harmless). A JSpecify-compatible tool will treat the two identically (apart from optionally
+ * reporting the redundancy itself).
+ *
+ * <p><b>Warning:</b> These annotations are under development, and <b>any</b> aspect of their
+ * naming, locations, or design is subject to change until the JSpecify 1.0 release. Moreover,
+ * supporting analysis tools will be tracking the changes on varying schedules. Releasing a library
+ * using these annotations in its API is <b>strongly discouraged</b> at this time.
+ *
+ * <h2>Details</h2>
+ *
+ * <p>An array of other annotation types may be provided, indicating that this annotation carries
+ * (at least) the combined meanings of <i>all</i> of them.
+ *
+ * <p>In the example above, if {@code Two} does not define any additional semantics, and lists only
+ * one implied annotation, we call it an <i>alias</i> for {@code One}. In this case, the annotations
+ * on <i>all</i> three example declarations above convey identical meaning, and should be treated
+ * identically by tools.
+ *
+ * <p>Implication is naturally transitive: if in addition to the example code above we also have
+ * {@code @Implies(Two.class) @interface Three {}}, it is understood that {@code Three} also implies
+ * {@code One}. Listing {@code One.class} explicitly is redundant but harmless.
+ *
+ * <p>It is not possible to supply custom attribute values for an implied annotation, such as
+ * declaring that {@code @Two} implies {@code @One("arg")} or that {@code @Two(key = VALUE)} implies
+ * {@code @One(key = VALUE)}.
+ *
+ * <p>The target types of {@code One} and {@code Two} may differ. {@code @Two} may therefore appear
+ * in a location where {@code @One} cannot; for these usages {@code @Implies} has no effect. The
+ * target types of {@code One} and {@code Two} might even be disjoint. For example if they are
+ * {@code {FIELD}} and {@code {TYPE_USE}}, there are still source locations where either can appear
+ * (before the type of a field), so the implication still has meaning.
+ *
+ * <p>If {@code Two} has wider retention than {@code One}, then in any situation where {@code @One}
+ * would not be visible, the effect of {@code @Implies} is undefined.
  */
 @Documented
 @Retention(RUNTIME)
 @Target(ANNOTATION_TYPE)
 public @interface Implies {
+  /**
+   * The annotation types implied by this annotation type
+   * (usually one). May not include any annotation type that
```

Comment:
I added it because I think it (subtly) gives the reader permission to focus on the simple case as they read, and deal with the possibility of multiples as an afterthought.

Implying multiple should be rare enough that I loathe the idea of it distracting anyone.

Reply:
I feel like adding the parenthetical calls attention to the possibility of multiple values, and raises the question of what the unusual cases are. But it's okay.

Reply:
Then I needed a different parenthetical so I nuked it anyway.